### PR TITLE
feat(unpaid invoice) add filter by service

### LINKED ISSUE #3731: Add filter by service to Unpaid Invoices Report
Often, our users would want to see the results of the unpaid invoices report for only a single service at a time.  Let's add this functionality for them by allowing them to select the service.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #3731

### CODE REVIEW

**Comment by kwilu** [comment]
Hello @jeremielodi, it seems like you have referenced #3731 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3731.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by kwilu** [comment]
Hello @jeremielodi, it seems like you have referenced #3731 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3731.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by kwilu** [comment]
Hello @jeremielodi, it seems like you have referenced #3731 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3731.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by kwilu** [comment]
Hello @jeremielodi, it seems like you have referenced #3731 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3731.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by kwilu** [comment]
Hello @jeremielodi, it seems like you have referenced #3731 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3731.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

File: server/models/procedures/invoicing.sql [hunk 1]
```diff
@@ -800,7 +800,7 @@ CREATE PROCEDURE UnbalancedInvoicePayments(
   -- this holds all the invoices that were made during the period
   -- two copies are needed for the UNION ALL query.
   DROP TABLE IF EXISTS tmp_invoices_1;
-  CREATE TABLE tmp_invoices_1 (INDEX uuid (uuid)) AS
+  CREATE TEMPORARY TABLE tmp_invoices_1 (INDEX uuid (uuid)) AS
```
Comment (mbayopanda) [inline server/models/procedures/invoicing.sql:803]: 👍

File: server/models/procedures/invoicing.sql [hunk 2]
```diff
@@ -809,10 +809,10 @@ CREATE PROCEDURE UnbalancedInvoicePayments(
     ORDER BY invoice.date;
 
   DROP TABLE IF EXISTS tmp_invoices_2;
-  CREATE TABLE tmp_invoices_2 AS SELECT * FROM tmp_invoices_1;
+  CREATE TEMPORARY TABLE tmp_invoices_2 AS SELECT * FROM tmp_invoices_1;
```
Comment (mbayopanda) [inline server/models/procedures/invoicing.sql:812]: 👍 good catch

File: server/controllers/finance/reports/unpaid-invoice-payments/index.js [hunk 3]
```diff
@@ -51,7 +51,11 @@ async function getUnbalancedInvoices(options) {
     new Date(options.dateTo),
   ];
 
-  const wherePart = options.debtorGroupName ? `WHERE debtorGroupName = ${db.escape(options.debtorGroupName)}` : '';
+  const { debtorGroupName, serviceId } = options;
+  let wherePart = debtorGroupName ? `WHERE debtorGroupName = ${db.escape(debtorGroupName)}` : '';
+  if (serviceId) {
+    wherePart = (wherePart.length < 2) ? `WHERE serviceID=${serviceId}` : `${wherePart} AND serviceId=${serviceId}`;
+  }
```
Comment (mbayopanda) [inline server/controllers/finance/reports/unpaid-invoice-payments/index.js:58]: 👍 very good

Is it possible to have on the view (handlebars) the selected service if it exists

**Review by mbayopanda — APPROVED** [review]
LGTM 👍

**Pull request merged**